lib/Installation/Partitioner/LibstorageNG/v4_3/ExpertPartitionerController.pm

```diff
@@ -121,6 +121,14 @@ sub add_partition_on_gpt_disk {
     $self->_add_partition($args->{partition});
 }
 
+sub add_partition_msdos {
+    my ($self, $args) = @_;
+    $self->get_expert_partitioner_page()->select_disk($args->{disk}) if $args->{disk};
+    $self->get_expert_partitioner_page()->press_add_partition_button();
+    $self->get_expert_partitioner_page()->select_new_partition_type($args->{partition_type});
```

Comment:
I added a subroutine, as otherwise the old one would fail while trying to use directly select_item_in_system_view_table instead of select_disk. I thought it could be better to have only one add_partition and to add "if $args->{partition_type}" at the end of this line, but then I'll need to touch other things. Please share thoughts.